kornia/geometry/camera/stereo.py

```diff
@@ -224,23 +243,24 @@ def _check_Q_matrix(Q_matrix: torch.Tensor):
         Q_matrix (torch.Tensor): The Q matrix for reprojecting disparity to a point cloud of shape :math:`(B, 4, 4)`
     """
     if not Q_matrix.ndim == 3:
-        raise StereoException(f"Expected 'Q_matrix to have 3 dimenstions." f"Got {Q_matrix.ndim}")
+        raise StereoException(f"Expected 'Q_matrix' to have 3 dimenstions." f"Got {Q_matrix.ndim}")
 
     if not Q_matrix.shape[1:] == (4, 4):
         raise StereoException(
             f"Expected last two dimensions of 'Q_matrix' to be of shape (4, 4)." f"Got {Q_matrix.shape}"
         )
 
-    if not Q_matrix.dtype == torch.float32:
-        raise StereoException(f"Expected 'Q_matrix' to be of type torch.float32." f"Got {Q_matrix.dtype}")
+    if Q_matrix.dtype not in (torch.float16, torch.float32, torch.float64):
+        raise StereoException(
+            f"Expected 'Q_matrix' to be of type torch.float16, torch.float32 or torch.float64." f"Got {Q_matrix.dtype}")
 
 
 def reproject_disparity_to_3D(disparity_tensor: torch.Tensor, Q_matrix: torch.Tensor) -> torch.Tensor:
     """
     Reproject the disparity tensor to a 3D point cloud.
 
     Args:
-        disparity_tensor (torch.Tensor): Disparity tensor of shape :math:`(B, H, W)`.
+        disparity_tensor (torch.Tensor): Disparity tensor of shape :math:`(B, 1, H, W)`.
         Q_matrix (torch.Tensor): Tensor of Q matrices of shapes :math:`(B, 4, 4)`.
 
     Returns:
```

Comment:
for consistency I'd return as image `BxHxWx3` - open to discussion

Reply:
@edgarriba  Do you mean `(B, H, W, 1)` or `(B, H, W, 3) i.e. a 3 channeled disparity image? I think channel last is fine, but a 3 channeled disparity map is odd.

Reply:
yeah `(B, H, W, 1)`